cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/DefaultKafkaQuotasManager.java

```diff
@@ -0,0 +1,210 @@
+                                }
+                            });
+                    }
+
+                    kafkaAdmin.close();
+                    promise.complete();
+
+                    return promise.future();
+                });
+        } catch (KafkaException e) {
+            LOGGER.warnCr(reconciliation, "Failed to alter default user quota", e);
+            promise.complete();
+
+            return promise.future();
+        }
+    }
+
+    /**
+     * Checks whether default quotas configuration needs to be altered with new configuration.
+     * It gets the current default quotas for users from Kafka and compares them with the desired configuration
+     *
+     * @param reconciliation          Reconciliation marker
+     * @param vertx                   Vert.x instance
+     * @param kafkaAdmin              Kafka Admin object
+     * @param ops                     List of {@link ClientQuotaAlteration.Op}
+     * @param isNotKafkaPlugin        boolean parameter determining if the Kafka built-in quotas plugin is used
+     *
+     * @return  result determining if the new quotas configuration should be applied or not
+     */
+    private static Future<Boolean> shouldAlterDefaultQuotasConfig(Reconciliation reconciliation, Vertx vertx, Admin kafkaAdmin, List<ClientQuotaAlteration.Op> ops, boolean isNotKafkaPlugin) {
+        return VertxUtil.kafkaFutureToVertxFuture(reconciliation,
+                vertx,
+                kafkaAdmin.describeClientQuotas(ClientQuotaFilter.containsOnly(List.of(ClientQuotaFilterComponent.ofDefaultEntity(ClientQuotaEntity.USER)))).entities())
+            .compose(clientQuotaEntityMapMap -> {
+                if (clientQuotaEntityMapMap != null) {
+                    Map<String, Double> currentQuotas = clientQuotaEntityMapMap.get(DEFAULT_USER_ENTITY);
+                    Map<String, Double> desiredQuotas = new HashMap<>();
+                    ops.forEach(op -> {
+                        // the result of `describeClientQuotas` contains just non-null values, in case that we will have
+                        // null values in desiredQuotas, the Maps will never be equal and the quotas will be altered every
+                        // reconciliation (even if the desired and current values will be same)
+                        if (op.value() != null) {
+                            desiredQuotas.put(op.key(), op.value());
+                        }
+                    });
+
+                    // 1. if the current quotas set in Kafka are null and quotas configuration in Kafka CR is not of type `QuotasPluginKafka`, skip alteration
+                    // 2. if the current quotas set in Kafka are null and quotas configuration in Kafka CR is type of `QuotasPluginKafka`, but there is no fields configured, skip alteration
+                    if ((currentQuotas == null && isNotKafkaPlugin) || (currentQuotas == null && desiredQuotas.isEmpty())) {
```

Comment:
Should you check it first before constructing the maps? Do you really need to compare the maps? Can't you just compare the 4 values?

Reply:
Comparison of two Maps was easier and the check if the `desiredQuotas` is empty is also easier with the Map.
But I understand that it's not that efficient as comparing just the values. However, the current quotas are in Map from the result, I cannot do much with that.